File: frontend/src/app/features/map/map.dto.ts
export interface TileDto {
  terrainId: number
  terrainTypeId : number
  accessible :number
  mapInteraction?: MapInteraction | null;
}

export interface MapDto {
  id: number;
  name: string;
  sightRange: number;
  clickRange: number;
  tiles: TileDto[][];
}

interface BaseInteraction {
  id: number
}

export interface ShopInteraction extends BaseInteraction {
  type: 'shop'
  shopName: string
}

export interface TeleporterInteraction extends BaseInteraction {
  type: 'teleporter'
  targetMapId: number
  targetX: number
  targetY: number
}

export type MapInteraction =
  | ShopInteraction
  | TeleporterInteraction

File: backend/Margorak.Api/Dto/MapDto.cs
namespace Margorak.Api.Dto
{
    public class MapDto
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public MapTileDto[][] Tiles {get;set;}
        public int SightRange { get; set; }
        public int ClickRange { get; set; }
    }
}
